src/detext/train/train_helper.py

```diff
@@ -13,30 +13,29 @@ def create_placeholder_for_ftrs(ph_name, shape, dtype, ftr_name, ftr_names, defa
     return ph, ph_one_batch
 
 
-def get_query(hparams, regex_replace_pattern, add_dimension=False):
+def get_query(hparams):
     """
     Helper function to get query and query_placeholder
     :param hparams: hparams
-    :param regex_replace_pattern: The regex pattern to add a white space before and after
-    :param add_dimension: whether to add a dimension then remove to query (this is to support online model for QAP as
-    quasar model serving requires at least one dimension)
     :return: query and query_placeholder
     """
     # query text feature
-    if add_dimension:
+    # If hparams.add_first_dim_for_query_placeholder is True, the query placeholder has dimension [None]
+    # This is to use the query feature as a document feature in model serving
+    if hparams.add_first_dim_for_query_placeholder:
         query_placeholder, query = create_placeholder_for_ftrs("query_placeholder", [None], tf.string, 'query',
                                                                hparams.feature_names)
     else:
         query_placeholder, query = create_placeholder_for_ftrs("query_placeholder", [], tf.string, 'query',
                                                                hparams.feature_names)
     if query is not None:
-        if add_dimension:
+        if hparams.add_first_dim_for_query_placeholder:
             # remove added dimension
             query = tf.squeeze(query, [0])
```

Comment:
I'm not very familiar with the serving logic which adds an additional dimension and squeezes it internally. Could you help me understand it? Does it only apply to string type input? (other folks may have the same questions)

Reply:
Thanks for the comment! 
Sure. The current problem is that for query and usr fields, the placeholders have dimension []. This causes some issues with our serving as it must have at least 1 dimension (dim 0, batch size). Therefore we add an option here to have the placeholders to have shape [None]. However, this would fail inside detext graph as it expects these inputs to have shape []. Therefore we have to squeeze it after the placeholder if it's been expanded.